### PR TITLE
[#11572] Notification Feature - Skeleton of DB

### PR DESCRIPTION
**Pull request by fsgmhoward**
Part of #11572 

**Outline of Solution**

This is part of the Notification Feature. This PR includes skeleton files for notification databases (classes for db, entity, and attributes).

This PR does not contain test cases but for the sake of keep development moving, it should be merged first. Test case will be added in a separate PR later.

### CODE REVIEW

File: src/main/java/teammates/logic/core/NotificationLogic.java [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+     * Gets notification associated with the {@code notificationId}.
+     *
+     * @return null if no match found.
+     */
+    public NotificationAttributes getNotification(String notificationId) {
+        return notificationDb.getNotification(notificationId);
+    }
+
+    /**
+     * Updates/Creates the profile using {@link StudentProfileAttributes.UpdateOptions}.
```
Comment (zhaojj2209) [inline src/main/java/teammates/logic/core/NotificationLogic.java:35]:
```suggestion
     * Updates/Creates the notification using {@link NotificationAttributes.UpdateOptions}.
```
Also, should we separate creating and updating the notification into two different functions? Doing so may be beneficial down the line if we were to add checks that apply only when updating the notification (e.g. disabling edit of title/message). It also doesn't really seem to make sense to throw an `EntityDoesNotExistExistException` here if we were to have the function handle both creating and updating of notification.

**Review by zhaojj2209 — CHANGES_REQUESTED** [review]

File: src/main/java/teammates/logic/core/NotificationLogic.java (hunk 1 above)
Comment (ziqing26) [reply src/main/java/teammates/logic/core/NotificationLogic.java:35]: I think the validations might be handled by the APIs when they receive relevant parameters as here is about the logic?

File: src/main/java/teammates/storage/entity/Notification.java [hunk 2]
```diff
@@ -0,0 +1,180 @@
...
+        this.setCreatedAt(Instant.now());
+
+        UUID uuid = UUID.randomUUID();
+        this.notificationId = uuid.toString();
+        this.shown = false;
+    }
+
+    /**
+     * Instantiates a new notification, with all fields passed in as parameters.
+     * This is mainly for convertion from attributes to entity.
```
Comment (ziqing26) [inline src/main/java/teammates/storage/entity/Notification.java:76]:
```suggestion
     * This is mainly for conversion from attributes to entity.
```

**Review by ziqing26 — CHANGES_REQUESTED** [review]
Just a small nitpick, otherwise LGTM!

File: src/main/java/teammates/logic/core/NotificationLogic.java (hunk 1 above)
Comment (fsgmhoward) [reply src/main/java/teammates/logic/core/NotificationLogic.java:35]: Thanks for the comment. This API is for update only. I forgot to put the create one inside. Will do so in next commit.
Yes, they are separate.

File: src/main/java/teammates/storage/entity/Notification.java (hunk 2 above)
Comment (fsgmhoward) [reply src/main/java/teammates/storage/entity/Notification.java:76]: Thanks for the comment.

**Review by ziqing26 — APPROVED** [review]
LGTM!

**Review by zhaojj2209 — APPROVED** [review]
Thanks for the changes, LGTM!

**Pull request merged**